pandas/core/arrays/datetimelike.py

```diff
@@ -80,6 +87,70 @@ def _get_attributes_dict(self):
         """
         return {k: getattr(self, k, None) for k in self._attributes}
 
+    @property
+    def _scalar_type(self):
+        """The scalar associated with this datelike
+
+        * PeriodArray : Period
+        * DatetimeArray : Timestamp
+        * TimedeltaArray : Timedelta
+        """
+        # type: # () -> Union[type, Tuple[type]]
+        raise AbstractMethodError(self)
+
+    def _scalar_from_string(self, value):
+        # type: (str) -> Union[Period, Timestamp, Timedelta, NaT]
+        """
+        Construct a scalar type from a string.
+
+        Parameters
+        ----------
+        value : str
+
+        Returns
+        -------
+        Period, Timestamp, or Timedelt, or NaT
+            Whatever the type of ``self._scalar_type`` is.
+        """
+        raise AbstractMethodError(self)
+
+    def _unbox_scalar(self, value):
```

Comment:
this doesn't really belong here, we have ``.value`` *on* the sclars themselves.

Reply:
Keep in mind that this is on the DatetimeLike base class. Each of the subclasses overrides with the appropriate behavior.

`DatetimeLikeArrayMixin.__setitem__` is a good example where all of these abstract methods are used. In this case, `_unbox_scalar` allows things like `PeriodArray[0] = Period('2000', 'D')`.

Reply:
the problem is that this is a scalar property and doesn't belong on the index. if you want to have an ``_unbox()`` method on the scalars be my guest, but on a container it doesn't  make sense.

Reply:
I don't follow that logic. This is analogous to `Index._box_func`.

Adding a `Period/Timedelta/Timestamp._unbox` method wouldn't work on it's own, since we'll still need to validate that the caller passed the correct type. We wouldn't want to accidentally unbox a Period on a TimedeltaArray.